src/posit/connect/resources.py

```diff
@@ -89,38 +93,50 @@ def update(self, **attributes):  # type: ignore[reportIncompatibleMethodOverride
         super().update(**result)
 
 
-T = TypeVar("T", bound=Resource)
+_T = TypeVar("_T", bound=Resource)
+_T_co = TypeVar("_T_co", bound=Resource, covariant=True)
```

Comment:
A perfect use case!

Linking for my own reminder: https://stackoverflow.com/a/61737894/591574

Reply:
But is it necessary since we're `cast()`ing?

I think we might be able to get away with a invariant type.

```suggestion
_R = TypeVar("_R", bound=Resource)
```

(`R` for factory _R_ eturn value?)